includes/Core/Modules/Module.php

```diff
@@ -319,29 +320,29 @@ public function is_connected() {
 				continue;
 			}
 
-			if ( empty( $datapoint_services[ $datapoint ] ) ) {
+			$datapoint_service = $datapoint_definitions[ $definition_key ]['service'];
+			if ( empty( $datapoint_service ) ) {
 				continue;
 			}
 
-			if ( ! isset( $service_batches[ $datapoint_services[ $datapoint ] ] ) ) {
-				$service_batches[ $datapoint_services[ $datapoint ] ] = $this->google_services[ $datapoint_services[ $datapoint ] ]->createBatch();
+			if ( ! isset( $service_batches[ $datapoint_service ] ) ) {
+				$service_batches[ $datapoint_service ] = $this->google_services[ $datapoint_service ]->createBatch();
 			}
 
-			$service_batches[ $datapoint_services[ $datapoint ] ]->add( $request, $key );
-			$results[ $key ] = $datapoint;
+			$service_batches[ $datapoint_service ]->add( $request, $key );
+			$results[ $key ] = $definition_key;
 		}
 
 		foreach ( $service_batches as $service_identifier => $batch ) {
 			try {
 				$batch_results = $batch->execute();
 			} catch ( Exception $e ) {
 				// Set every result of this batch to the exception.
-				foreach ( $results as $key => $datapoint ) {
-					if ( ! is_string( $datapoint ) || ! isset( $datapoint_services[ $datapoint ] ) || $service_identifier !== $datapoint_services[ $datapoint ] ) {
-						continue;
+				foreach ( $results as $key => $definition_key ) {
+					$datapoint_service = ! empty( $datapoint_definitions[ $definition_key ] ) ? $datapoint_definitions[ $definition_key ]['service'] : null;
+					if ( is_string( $definition_key ) && $service_identifier === $datapoint_service ) {
+						$results[ $key ] = $this->exception_to_error( $e, $definition_key );
```

Comment:
The second argument to `exception_to_error` is still expected to be the datapoint name (i.e. `$dataset->datapoint`). I think the key here would be more useful but we probably shouldn't change that in this issue.

Same below on `364`.

Reply:
Fixed.